pyanaconda/payload/dnf/payload.py

```diff
@@ -1986,6 +1986,17 @@ def _write_dnf_repo(self, repo, repo_path):
             if ks_repo.excludepkgs:
                 f.write("exclude=%s\n" % ",".join(ks_repo.excludepkgs))
 
+    def _import_rpm_keys(self):
```

Comment:
I think that you don't need this function. You need only properly setup path in DNF repo object.

Reply:
It could be replaced by `dnf.Base().package_import_key()`. Also check `dnf.Base().package_signature_check()` for detection if import is needed.

Reply:
This could be far better. Thank you for the pointers!

Reply:
@j-mracek does this take the certs from chroot or host?

Reply:
@j-mracek, we are trying to import gpg keys that are installed on the target system by rpm packages. I don't see how we could use `dnf.Base()` configured by the installation environment that might use different keys and different repositories than the installed ones.

Reply:
I suggest that keys used for verification are taken from RPM_DB from chroot and also imported to the same DB. The location of source keys for import are taken from repo.gpgkey (see all repos from base.repos). Paths or urls in repo.gpgkey are not related or modified according to chroot.

Reply:
> eys that are installed on the target system by rpm packages. I don't see how we could use `dnf.Base()` configured by the installation environment that might use different keys and different

We can discuss it on call if you like. In case that you want to install, upgrade packages, rpm verify it if the gpg key is valid and inserted in rpm_db. DNF provide a functionality how to verify and import missing gpg keys for a package. You only need to set properly path in dnf.repo.Repo() class using urllist attribute `gpgkey`.

Both methods `dnf.Base().package_import_key()` and `dnf.Base().package_signature_check()` are brand new APIs to allow user of DNF (anaconda, dnfdeamon) to not have their own implementation of gpg import mechanism.